assets/app/scripts/controllers/deployment.js

```diff
@@ -127,11 +109,22 @@ angular.module('openshiftConsole')
           });
         };
 
-        var limitRanges = {};
-
-        var checkCPURequest = function() {
-          var containers = _.get($scope, 'deployment.spec.template.spec.containers', []);
-          $scope.hasCPURequest = HPAService.hasCPURequest(containers, limitRanges, project);
+        var allHPA = {}, limitRanges = {};
+        var updateHPA = function() {
+          $scope.hpaForRC = HPAService.hpaForRC(allHPA, $routeParams.deployment || $routeParams.replicationcontroller);
+          if ($scope.isActive) {
+            // Show both HPAs that target the RC and the DC if this is the active deployment.
+            var hpaForDC = HPAService.hpaForDC(allHPA, $routeParams.deploymentconfig);
+            $scope.autoscalers = $scope.hpaForRC.concat(hpaForDC);
+          } else {
+            $scope.autoscalers = $scope.hpaForRC;
+          }
+        };
+        var updateHPAWarnings = function() {
+            HPAService.getHPAWarnings($scope.deployment, $scope.autoscalers, limitRanges, project)
+                      .then(function(warnings) {
+              $scope.hpaWarnings = warnings;
```

Comment:
when the promise returns immediately i can see this updating correctly but what would happen if the promise is actually async?  do we need an evalAsync or something here